core/objectstore/objectstore.go

```diff
@@ -73,6 +73,13 @@ type ObjectStoreGetter interface {
 	GetObjectStore(context.Context, string) (ObjectStore, error)
 }
 
+// SingularObjectStoreGetter is the interface that is used to get a singular
```

Comment:
So this is a bit annoying now we're dropping the word Factory. We have a way to get an ObjectStore based on a model UUID. That's called ObjectStoreGetter. We also want to late bind the getting of an ObjectStore for two reasons:

 1. We can't return an error early on to get the ObjectStore, as that requires the changing of the ServiceFactory method. This then unfortunately would force everyone to write `if err != nil {` in places it doesn't really make sense.
 2. Lazy evaluation then provides opportunities to only access the ObjectStore when strictly necessary.

The other restriction is that we don't want to provide a way to get other model ObjectStores. We know the scope (model UUID) of the model because of the way we've setup the domain services. So we need a singular access to the ObjectStoreGetter. That's how I arrived at that name.

Essentially we need a name for this that isn't called ObjectStoreGetter.

```
type ??? interface {
    GetObjectStore(context.Context) (ObjectStore, error)
}
```

Reply:
We settled on ModelObjectStoreGetter as it clearly defines expectations on the resulting ObjectStore.